### PR TITLE
Decompiles z_bg_spot11_bakudankabe.c

### PR DESCRIPTION
**Pull request by i82orbom**

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Spot11_Bakudankabe/z_bg_spot11_bakudankabe.c [hunk 1]
```diff
@@ -15,7 +15,8 @@ void BgSpot11Bakudankabe_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void BgSpot11Bakudankabe_Update(Actor* thisx, GlobalContext* globalCtx);
 void BgSpot11Bakudankabe_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-/*
+void func_808B2180(BgSpot11Bakudankabe* this, GlobalContext* globalCtx);
+
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Spot11_Bakudankabe/z_bg_spot11_bakudankabe.c:19]:
```suggestion
```

File: src/overlays/actors/ovl_Bg_Spot11_Bakudankabe/z_bg_spot11_bakudankabe.c [hunk 2]
```diff
@@ -27,15 +28,117 @@ const ActorInit Bg_Spot11_Bakudankabe_InitVars = {
...
+extern Gfx D_0500A880[];
+
+void func_808B2180(BgSpot11Bakudankabe* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    this->collider.dim.pos.x += ((s16)this->dyna.actor.posRot.pos.x);
+    this->collider.dim.pos.y += ((s16)this->dyna.actor.posRot.pos.y);
+    this->collider.dim.pos.z += ((s16)this->dyna.actor.posRot.pos.z);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Spot11_Bakudankabe/z_bg_spot11_bakudankabe.c:52]:
```suggestion
    this->collider.dim.pos.x += (s16)this->dyna.actor.posRot.pos.x;
    this->collider.dim.pos.y += (s16)this->dyna.actor.posRot.pos.y;
    this->collider.dim.pos.z += (s16)this->dyna.actor.posRot.pos.z;
```

**Review by Roman971 — COMMENTED** [review]
Make sure to run the format script on this file

**Comment by i82orbom** [comment]
> Make sure to run the format script on this file

Uhmm, normally I do using clang-format from vscode, these parentheses weren't actually picked up either with the format script, should have they?

**Comment by Roman971** [comment]
That formatting mention wasn't about the comments I made. But I noticed the file didn't have a final line break, which is handled by the format script.

**Comment by fig02** [comment]
i use vscode clang-format too, which wont do things like add final newline. Thats handled by clang-tidy I believe.
So yeah just run the whole script right before a PR even if you use the vscode plugin for stuff while doing the file

**Comment by Roman971** [comment]
Final new line is a manual step done by the format script itself.

**Review by Roman971 — APPROVED** [review]

File: src/overlays/actors/ovl_Bg_Spot11_Bakudankabe/z_bg_spot11_bakudankabe.c [hunk 3]
```diff
@@ -27,15 +26,117 @@ const ActorInit Bg_Spot11_Bakudankabe_InitVars = {
...
+
+    sp24 = 0;
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F))) {
+        Actor_Kill(&this->dyna.actor);
+        return;
+    }
+    func_808B2180(this, globalCtx);
+    DynaPolyInfo_Alloc(&D_06001A58, &sp24);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, (Actor*)this, sp24);
```
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Spot11_Bakudankabe/z_bg_spot11_bakudankabe.c:112]: &this->dyna.actor instead of casting

**Pull request merged**